### ISSUE TITLE
Metropolis not thread safe?

### ISSUE THREAD

**Issue by simone-romiti** [urbach/su2#55]
If we run the metropolis with different numbers of cores the result changes.
The issue lies in the `sweep` function. Parallelizing in "t" only is not enough, as we should account also of nearest enighbors in the spatial directions.

**Comment by urbach** [comment]
> If we run the metropolis with different numbers of cores the result changes.
> The issue lies in the `sweep` function. Parallelizing in "t" only is not enough, as we should account also of nearest enighbors in the spatial directions.

one needs to do even/odd in t-direction, but that should be enough.

what do you mean by "the results change?" They are not expected to be
identical, are they?

**Comment by christianegross** [comment]
By how much do the results change? I do not expect it to be exactly identical, because the number of random number generators we assign and use depends on the number of cores, and I am not sure if the loop iterations are distributed statically or dynamically to the threads.

**Comment by simone-romiti** [comment]
Yes, I mean "notidentical". They are fairly similar though. I am currently trying to implement a thread safe version of that.
